src/MassTransit.MessagePack/Serialization/MassTransitMessagePackFormatterResolver.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace MassTransit.Serialization;
+
+using System;
+using System.Collections.Generic;
+using Internals;
+using MessagePack;
+using MessagePack.Formatters;
+using MessagePackFormatters;
+
+
+class MassTransitMessagePackFormatterResolver : IFormatterResolver
+{
+    public static MassTransitMessagePackFormatterResolver Instance { get; } = new();
+
+    readonly IReadOnlyCollection<KeyValuePair<Type, Type>> _mappedFormatterByType;
+    readonly IDictionary<Type, IMessagePackFormatter> _cachedFormatters;
+
+    private MassTransitMessagePackFormatterResolver()
+    {
+        _mappedFormatterByType = new List<KeyValuePair<Type, Type>>
+        {
+            new(typeof(MessageData<>), typeof(MessageDataFormatter<>)),
+        };
+
+        _cachedFormatters = new Dictionary<Type, IMessagePackFormatter>
```

Comment:
Any reason you haven't included all of the concrete types of the interface message types that are used by STJ?
https://github.com/MassTransit/MassTransit/blob/develop/src/MassTransit/Serialization/JsonConverters/SystemTextJsonConverterFactory.cs#L34

Reply:
The one mapping you spotted was left by mistake after `InterfaceMessagePackFormatter` was introduced.
But with that said, in the interest of limiting reflection where possible and behaving as close to `SystemTextJsonSerializer` as possible, **I have added mappers to the concrete types you referenced**.